PYME/recipes/kinetics_recipes.py

```diff
@@ -0,0 +1,428 @@
+
+        pylab.show()
+
+        powers = np.array([res_on[0][1], res_off[0][1]])
+        times = np.array([cyctime, (max(ds[oncol]).astype(float)+1)*cyctime, (max(ds[offcol]).astype(float)+1)*cyctime])
+
+        output_list = [powers, times, n_on, n_off]
+
+        namespace[self.outputEmphistData] = output_list
+
+@register_module('save_EmpHistJSON')
+class save_EmpHistJson(ModuleBase):
+    """
+    Creates empirical histogram files for use with the simulator, based on the outputs from the fitting module above.
+    """
+    inputName = Input('emphist_data')
+    outputdir = CStr('C:\\Users\\bdr25\\Desktop\\data\\JSON files')
+    outputName = Output('emphist_file')
+
+    def execute(self, namespace):
+        import io
+        import json
+        import time
+
+        powers, times, n_on, n_off = namespace[self.inputName]
+
+        new_on_nums = np.zeros(len(n_on)+1)
+        new_off_nums = np.zeros(len(n_off)+1)
+        new_on_nums[1:] = n_on
+        new_off_nums[1:] = n_off
+
+        json_prepped_on = np.flip((new_on_nums.reshape((len(new_on_nums), 1))).astype('int'), axis=1).tolist()
+        json_prepped_off = np.flip((new_off_nums.reshape((len(new_off_nums), 1))).astype('int'), axis=1).tolist()
+
+        dk = {}
+        dk['plog'] = [0, 0]
+        dk['tlog'] = [0, 0]
+        dk['pmin'] = [min(powers), min(powers)]
+        dk['pmax'] = [max(powers), max(powers)]
+        dk['tmin'] = [times[0], times[0]]
+        dk['tmax'] = [times[1], times[2]]
+        dk['on'] = json_prepped_on
+        dk['off'] = json_prepped_off
+
+        output_dir = self.outputdir
+
+        try:
+            to_unicode = unicode
+        except NameError:
+            to_unicode = str
+
+        timestr = time.strftime("%Y%m%d-%H%M%S")
+
+        with io.open(output_dir + '\\' + 'test_emphist' + timestr + '.json', 'w', encoding='utf8') as outfile:
+            str_ = '{"dk": ' + json.dumps(dk) + '}'
+            outfile.write(to_unicode(str_))
+
+        print('Json has been created and saved to disk')
+
+
+@register_module('HDBSCANClustering')
+class HDBSCANClustering(ModuleBase):
+    """
+    Performs HDBSCAN clustering on input dictionary
+
+    Parameters
+    ----------
+
+        minPtsForCore: The minimum size of clusters. Technically the only required parameter.
+        searchRadius: Extract DBSCAN clustering based on search radius. Skipped if 0 or None.
+
+    Notes
+    -----
+
+    See https://github.com/scikit-learn-contrib/hdbscan
+    Lots of other parameters not mapped.
+
+    """
+
+    input_name = Input('filtered')
+    columns = ListStr(['x', 'y'])
+    search_radius = Float()
+    min_clump_size = Int(100)
+    clump_column_name = CStr('hdbscan_id')
+    clump_prob_column_name = CStr('hdbscan_prob')
+    clump_dbscan_column_name = CStr('dbscan_id')
+    output_name = Output('hdbscan_clustered')
+
+    def execute(self, namespace):
+
+        inp = namespace[self.input_name]
+        mapped = tabular.mappingFilter(inp)
```

Comment:
`mappingFilter` is deprecated in favor of `MappingFilter` -> you'll want to run test this all again merged into latest PYME head and fix any deprecation warnings